### PR TITLE
feat: 刷选时支持高亮所有对应的行列头cell

### LINKED ISSUE #1667: 🙏框选时高亮所有对应的行列头单元格
### 🖋 Description
Current：
image

对比单个选中时：
image

Expected:
image

### 🧐 Suggest the API

|  name  |  type  | default  | description |
|----|----|----|----|
|  -  |  -  |  -  |  -  |

### PR DESCRIPTION
**Pull request by serializedowen**
### 👀 PR includes

✨ Feature

- [x] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [x] Solve the issue and close #1667 

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description
Before：selectedHighlight仅作用于data-cell-click事件，导致刷选情况无法触发高亮行列头
image

将高亮对应行列头function抽离至
https://github.com/antvis/S2/blob/c5335badf47ccc89246c5a6d0575395bbb453f36/packages/s2-core/src/utils/interaction/select-event.ts

在brush-selection中调用

After: 刷选会更新所有的对应cell
image

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

**Comment by lijinke666** [comment]
为啥图片上看起来只高亮了序号列?

File: packages/s2-core/src/utils/interaction/select-event.ts [hunk 1]
```diff
@@ -81,3 +85,20 @@ export function getRowCellForSelectedCell(
     spreadsheet.isHierarchyTreeType(),
   );
 }
+
+export function updateRowColCells(meta: ViewMeta, spreadsheet: SpreadSheet) {
```
Comment (lijinke666) [inline packages/s2-core/src/utils/interaction/select-event.ts:89]: ViewMeta 里面有 `spreadsheet` 不用单独传一个参数

**Comment by lijinke666** [comment]
文档和 API 都记得处理下

https://github.com/antvis/S2/blob/3563f3c9a48998827722babb265e7889b7a86a20/s2-site/docs/manual/advanced/interaction/basic.zh.md#L165-L167

对于透视表是 work 的吗? 没看到相关单测

File: packages/s2-core/src/utils/interaction/select-event.ts (hunk 1 above)
Comment (serializedowen) [reply packages/s2-core/src/utils/interaction/select-event.ts:89]: done

**Comment by serializedowen** [comment]
列头有高亮的，目前没有针对透视表测试过，可能存在问题，单测补充在brush-selection-spec里了

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**